### PR TITLE
#10 - implement json logging support

### PR DESCRIPTION
**Pull request by onmomo**
This will present the logs as follows if configured as shown in the examples.

```
Starting ECR-updater ..
Loading log configuration file from path './logging.json'...
{"asctime": "2021-05-31 08:13:37,798", "levelname": "INFO", "message": "Running credentials update loop ..", "pathname": "ecrupdater.py", "lineno": 178, "threadName": "MainThread"}
{"asctime": "2021-05-31 08:13:37,798", "levelname": "INFO", "message": "Starting ECR secret update loop ..", "pathname": "ecrupdater.py", "lineno": 97, "threadName": "MainThread"}
{"asctime": "2021-05-31 08:13:38,892", "levelname": "WARNING", "message": "NoCredentialsError: Unable to locate credentials", "pathname": "ecrupdater.py", "lineno": 30, "threadName": "MainThread"}
```
This way the logs can easily be parsed on a log aggregator such as graylog or ELK.

AC
- [x] support optional json logging. Will modify the current stdout slightly.
- [x] bump python to 3.8
- [x] lock pypi dependencies

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -6,8 +6,9 @@ verify_ssl = true
 [dev-packages]
 
 [packages]
-boto3 = "*"
-kubernetes = "==3.0.*"
+boto3=">=1.17.0,<1.18"
+kubernetes=">=17.17.0,<18.0"
+python-json-logger="1.3.0"
```
Comment (timothee-starmind) [inline Pipfile:11]: We used this library recently: https://github.com/bobbui/json-logging-python
It's also the one mentioned in Gitbooks: https://app.gitbook.com/@starmind/s/internal/the-engineers-handbook/logging/logging-in-python

Not sure which one is best, but the one from gitbooks seems to be maintained

File: ecrupdater.py [hunk 2]
```diff
@@ -4,24 +4,42 @@
...
+        with open(log_config_file, 'rt') as f:
+            config = json.load(f)
+        logging.config.dictConfig(config)
+    else:
+        print("No Log config file provided, loading default configuration.")
+        logging.basicConfig(level=default_level)
 
 
 def exception_handler(exception_type, exception, traceback):
     # Removes standard python stacktrace in case of exceptions
```
Comment (timothee-starmind) [inline ecrupdater.py:29]: Is this a wanted behaviour to remove the stacktrace? (If yes LGTM)

**Review by timothee-starmind — APPROVED** [review]
Look good to me!

I would double check that the library is well maintained, or use https://github.com/bobbui/json-logging-python to be consistent with gitbooks

It looks from the output that the `sm_logger` field will not get populated. It's optional, and I think we only have one main module so it wouldn't be too useful at this point. But if you want to add it in the future, you probably need to add `%(name)s` to your format string if I'm correct 😄

File: ecrupdater.py (hunk 2 above)
Comment (onmomo) [reply ecrupdater.py:29]: I don't know why they want to do this actually..

Comment (timothee-starmind) [reply ecrupdater.py:29]: OK well, it was like this before so 🤷

**Pull request merged**